go/libraries/doltcore/diff/async_differ.go

```diff
@@ -59,18 +60,20 @@ type AsyncDiffer struct {
 	eg       *errgroup.Group
 	egCtx    context.Context
 	egCancel func()
+
+	diffStats map[types.DiffChangeType]uint64
 }
 
 var _ RowDiffer = &AsyncDiffer{}
 
 // todo: make package private once dolthub is migrated
 func NewAsyncDiffer(bufferedDiffs int) *AsyncDiffer {
 	return &AsyncDiffer{
-		make(chan diff.Difference, bufferedDiffs),
-		bufferedDiffs,
-		nil,
-		context.Background(),
-		func() {},
+		diffChan:   make(chan diff.Difference, bufferedDiffs),
+		bufferSize: bufferedDiffs,
+		egCtx:      context.Background(),
+		egCancel:   func() {},
```

Comment:
do we need to initialize these?

Reply:
I just kept this because it was there?  I assume it's being called later  without being nil checked.